File: challenge-167/michael-dicicco/python/task1.py
# Challenge 167 Task1
#
#  Solution By: Michael DiCicco
#
#  Write a script to find out first 10 circular primes having at least 3
#  digits (base 10). Please checkout wikipedia for more information. A
#  circular prime is a prime number with the property that the number
#  generated at each intermediate step when cyclically permuting its (base
#  10) digits will also be prime.
#
#  Output 113, 197, 199, 337, 1193, 3779, 11939, 19937, 193939, 199933

class CircularPrime:
    used_numbers = set()

    @staticmethod
    def is_prime(some_number):
        if int(some_number) <= 1:
            return False
        for i in range(2, int(some_number) - 1):
            if int(some_number) % i == 0:
                return False
        return True

    @staticmethod
    def rotate(some_number):
        return some_number[1: len(some_number)] + some_number[0]

    @staticmethod
    def validate(some_number):
        if not CircularPrime.is_prime(some_number):
            return False
        tmp = some_number
        for i in range(0, len(some_number) - 1):
            tmp = CircularPrime.rotate(tmp)
            if not CircularPrime.is_prime(int(tmp)):
                return False
            CircularPrime.used_numbers.add(tmp)
        return True


def main():
    ten_circular_primes = []
    some_number = "111"
    while len(ten_circular_primes) < 10:
        if CircularPrime.validate(some_number):
            if some_number not in CircularPrime.used_numbers:
                ten_circular_primes.append(some_number)
        some_number = str(int(some_number) + 1)
    print(", ".join(ten_circular_primes))


if __name__ == '__main__':
    main()

File: challenge-167/michael-dicicco/java/challenge167/task1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package challenge167;

/**
 *
 * @author mddicicco
 */
public class task1 {

    /**
     * Challenge 167 Task1
     *
     * Solution By: Michael DiCicco
     *
     * Write a script to find out first 10 circular primes having at least 3
     * digits (base 10). Please checkout wikipedia for more information. A
     * circular prime is a prime number with the property that the number
     * generated at each intermediate step when cyclically permuting its (base
     * 10) digits will also be prime.
     *
     * Output 113, 197, 199, 337, 1193, 3779, 11939, 19937, 193939, 199933
     */
    
    public static void main(String[] args) {
        String[] ten_circular_primes = new String[10];
        String someNumber = "111";
        int index = 0;
        while (ten_circular_primes[9] == null) {
            if (circularprime.validate(someNumber)) {
                if (!circularprime.used_numbers.contains(someNumber)) {
                    ten_circular_primes[index] = someNumber;
                    index++;
                }
            }
            someNumber = Integer.toString(Integer.parseInt(someNumber) + 1);
        }
        System.out.println(String.join(", ", ten_circular_primes));

    }
}
